DependencyInjection/Compiler/DescriptionEnhancerPass.php

```diff
@@ -59,12 +70,26 @@ public function process(ContainerBuilder $container)
             ));
         }
 
+        // Remove inactive enhancers from the container.
+        //
+        // Note that previously we relied on the container automatically removing
+        // unreferenced definitions, but this proved to not always be the case. It
+        // is safer to remove them here.
         $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
```

Comment:
This is where the problem arises: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/CheckExceptionOnInvalidReferenceBehaviorPass.php#L34

Which processes all the definitions and checks for invalid references.